message.go

```diff
@@ -36,6 +36,58 @@ const (
 )
 
 type Props map[string]string
+
+func (p Props) GetIntValue(key string) (int64, bool, error) {
+	val, ok := p[key]
+	if !ok {
+		return 0, false, fmt.Errorf("property %s not found", key)
+	}
+
+	i, err := strconv.ParseInt(val, 10, 64)
+	if err != nil {
+		return 0, true, fmt.Errorf("property %s has wrong value type. expected int, got %s", key, val)
+	}
+
+	return i, true, nil
+}
+
+func (p Props) GetStringValue(key string) (string, error) {
+	val, ok := p[key]
+	if !ok {
+		return "", fmt.Errorf("property %s not found", key)
+	}
+
+	return val, nil
+}
+
+func (p Props) GetFloatValue(key string) (float64, bool, error) {
+	val, ok := p[key]
+	if !ok {
+		return 0, false, fmt.Errorf("property %s not found", key)
```

Comment:
If you introduced bool for presence check, do not return error here. Just return 0, false, nil.